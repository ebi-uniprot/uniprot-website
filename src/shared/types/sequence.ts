export type Sequence = {
  value: string;
  length: number;
  molWeight: number;
  /**
   * @deprecated Should use MD5 instead
   */
  crc64: string;
  md5: string;
};
