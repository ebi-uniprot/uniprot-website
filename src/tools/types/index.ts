export enum Tool {
  blast = 'blast',
  // align = 'align',
  // peptideSearch = 'peptideSearch',
  // idMapping = 'idMapping',
}
