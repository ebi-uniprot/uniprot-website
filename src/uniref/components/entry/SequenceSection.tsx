import React, { FC } from 'react';
import { Card, Sequence } from 'franklin-sites';

import { formatLargeNumber, hasContent } from '../../../shared/utils/utils';

import EntrySection, { EntrySectionIDs } from '../../types/entrySection';

import { SequenceUIModel } from '../../../uniprotkb/adapters/sequenceConverter';

const SequenceSection: FC<{
  data: SequenceUIModel;
  primaryAccession: string;
}> = ({ data }) => {
  if (!hasContent(data)) {
    return null;
  }

  const infoData = [
    {
      title: 'Length',
      content: data.sequence.length,
    },
    {
      title: 'Mass (Da)',
      content: formatLargeNumber(data.sequence.molWeight),
    },
    {
      title: 'Checksum',
      content: data.sequence.crc64,
    },
  ];

  return (
    <div id={EntrySectionIDs[EntrySection.Sequence]} data-entry-section>
      <Card title={EntrySection.Sequence}>
        <Sequence
          sequence={data.sequence.value}
          infoData={infoData}
          isCollapsible
        />
      </Card>
    </div>
  );
};

export default SequenceSection;
