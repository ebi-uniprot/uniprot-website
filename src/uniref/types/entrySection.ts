enum EntrySection {
  Members = 'Members',
  Sequence = 'Representative sequence',
}

export default EntrySection;

export const EntrySectionIDs = {
  [EntrySection.Members]: 'members',
  [EntrySection.Sequence]: 'sequence',
};
