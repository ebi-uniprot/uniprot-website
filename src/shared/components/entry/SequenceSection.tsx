import { FC } from 'react';
import { Card, Sequence } from 'franklin-sites';

import { Sequence as SequenceType } from '../../types/sequence';
import { formatLargeNumber, hasContent } from '../../utils/utils';

import EntrySection, {
  getEntrySectionNameAndId,
} from '../../../uniref/types/entrySection';

const SequenceSection: FC<{
  data: SequenceType;
}> = ({ data }) => {
  if (!hasContent(data)) {
    return null;
  }

  const infoData = [
    {
      title: 'Length',
      content: data.length,
    },
    {
      title: 'Mass (Da)',
      content: formatLargeNumber(data.molWeight),
    },
    {
      title: 'Checksum',
      content: data.crc64,
    },
  ];

  return (
    <div id={EntrySection.Sequence}>
      <Card title={getEntrySectionNameAndId(EntrySection.Sequence).name}>
        <Sequence sequence={data.value} infoData={infoData} isCollapsible />
      </Card>
    </div>
  );
};

export default SequenceSection;
