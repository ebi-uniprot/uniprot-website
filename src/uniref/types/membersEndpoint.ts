import { SearchResults } from '../../shared/types/results';
import { UniRefMember } from '../adapters/uniRefConverter';

export type UniRefMembersResults = SearchResults<UniRefMember>;

export const uniRefMembersFacets = [
  'member_id_type',
  'uniprot_member_id_type',
] as const;
