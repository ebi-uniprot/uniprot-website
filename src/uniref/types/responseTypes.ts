import { UniRefLiteAPIModel } from '../adapters/uniRefConverter';
import { FacetObject } from '../../uniprotkb/types/responseTypes';

type Response = {
  data: {
    results: UniRefLiteAPIModel[];
    facets?: FacetObject[];
  };
  headers: {
    ['x-totalrecords']: string;
    link: string;
  };
};

export default Response;
