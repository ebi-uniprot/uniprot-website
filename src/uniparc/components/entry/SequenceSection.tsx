import { Card } from 'franklin-sites';

import SimpleSequence from '../../../shared/components/simple-sequence/SimpleSequence';

import { hasContent } from '../../../shared/utils/utils';
import { getEntrySectionNameAndId } from '../../utils/entrySection';

import EntrySection from '../../types/entrySection';
import { Sequence as SequenceType } from '../../../shared/types/sequence';

const SequenceSection = ({ data }: { data?: SequenceType }) => {
  if (!data || !hasContent(data)) {
    return null;
  }

  return (
    <Card
      header={<h2>{getEntrySectionNameAndId(EntrySection.Sequence).name}</h2>}
      id={EntrySection.Sequence}
    >
      <SimpleSequence sequence={data} />
    </Card>
  );
};

export default SequenceSection;
