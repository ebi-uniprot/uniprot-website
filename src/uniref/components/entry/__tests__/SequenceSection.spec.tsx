import { render } from '@testing-library/react';

import SequenceSection from '../SequenceSection';

import { Sequence as SequenceType } from '../../../../shared/types/sequence';

import sequenceUIData from '../../../../shared/components/entry/__tests__/__mocks__/sequenceUIData.json';

describe('SequenceSection component', () => {
  test('should render the SequenceSection properly and match snapshot', () => {
    const { asFragment } = render(
      <SequenceSection data={sequenceUIData.sequence as SequenceType} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should return null when the provided object values are empty', () => {
    const { asFragment } = render(
      <SequenceSection data={{ value: undefined } as SequenceType} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
