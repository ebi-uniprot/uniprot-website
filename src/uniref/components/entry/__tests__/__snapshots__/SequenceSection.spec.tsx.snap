// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SequenceSection component should render the SequenceSection properly and match snapshot 1`] = `
<DocumentFragment>
  <div
    id="sequence"
  >
    <div
      class="card"
    >
      <section>
        <div
          class="card__header"
        >
          <h2
            class="card__title"
          >
            Representative
          </h2>
        </div>
        <div
          class="card__content"
        >
          <button
            class="button secondary"
            type="button"
          >
            Show sequence 
          </button>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SequenceSection component should return null when the provided object values are empty 1`] = `<DocumentFragment />`;
